src/local/experiment.py

```diff
@@ -231,12 +234,69 @@ def eval_and_log(self):
         mlflow.log_metric('val_r2', r2_val)
         print(f"Test r2: {r2_test}")
         mlflow.log_metric('test_r2', r2_test)
+
+
+class LassoNetExperiment(NNExperiment):
+
+    def create_model(self):
+        self.model = LassoNetRegressor(
+            input_size=self.X_train.shape[1],
+            hidden_size=self.cfg.model.hidden_size,
+            optim_params=self.cfg.experiment.optimizer,
+            scheduler_params=self.cfg.experiment.scheduler,
+            alpha_start=self.cfg.model.alpha_start,
+            alpha_end=self.cfg.model.alpha_end,
+            init_limit=self.cfg.model.init_limit
+        )
+
+    def load_best_model(self):
+        self.model = LassoNetRegressor.load_from_checkpoint(
+            self.trainer.checkpoint_callback.best_model_path,
+            input_size=self.X_train.shape[1],
+            hidden_size=self.cfg.model.hidden_size,
+            optim_params=self.cfg.experiment.optimizer,
+            scheduler_params=self.cfg.experiment.scheduler,
+            alpha_start=self.cfg.model.alpha_start,
+            alpha_end=self.cfg.model.alpha_end,
+            init_limit=self.cfg.model.init_limit
+        )
+
+    def eval_and_log(self):
+        self.model.eval()
+        train_preds, val_preds, test_preds = self.trainer.predict(self.model, self.data_module)
+        
+        train_preds = torch.cat(train_preds).squeeze().cpu().numpy()
+        val_preds = torch.cat(val_preds).squeeze().cpu().numpy()
+        test_preds = torch.cat(test_preds).squeeze().cpu().numpy()
+        
+        print(train_preds.shape, val_preds.shape, test_preds.shape)
+        
+        best_train_r2 = -1e+3
+        best_val_r2 = -1e+3
+        best_test_r2 = -1e+3
+        best_col = None
+        for col in range(val_preds.shape[1]):
```

Comment:
Not clear what you're doing here. Please add a comment and/or docs to the method